### PR TITLE
improving docs and varia

### PR DESCRIPTION
**Pull request by jposada202020**
improving docs
including learning guides, and ICM20948 breakout in hardware API. 
Including all examples in API. 
Re-arranging class doc order in api.rst

Built locally to verify

### CODE REVIEW

**Review by dhalbert — COMMENTED** [review]
@jposada202020 Thank you for doing these improvements to multiple libraries. 

A couple of comments which apply to most of these PR's:

The phrasing _"Here is one way of importing the `MCP9600` class so you can use it with the name ``mcp``"_, and similar sort of suggest to me there's something special about the name `mcp`. I might rewrite this as  _"Here is an example of creating and using an `MCP9600` object, assigning it to the variable ``mcp``_." Or even just _"Here is an example of using the `MCP9600` class."_

We often recommend people use `board.I2C()` instead of `busio.I2C(board.SCL, board, SDA)`. It's a common support issue that people try to create multiple instances of `busio.I2C(board.SCL, board, SDA)` and are confused by the errors on the second try. The advantage of using `board.I2C()` is that this problem is avoided. The disadvantage is that it's not clear which pins are being used, though that could be in a comment, e.g.:
```py
i2c = board.I2C()    # Uses board.SCL and board.SDA
```

Many of the simpletests were written before `board.I2C()` existed. @kattni, do you have a comment about whether `board.I2C()` or `busio.I2C()` would be preferred for these simple examples?

**Comment by jposada202020** [comment]
@dhalbert Thanks for the comments. good points
1. I will modify the phrasing according to your recommendations. Thanks
2. Regarding the use of busio.  A lot of the examples use busio.  A lot of learning guides use it also in the example code.
### One Example
https://learn.adafruit.com/lsm6ds33-6-dof-imu=accelerometer-gyro/python-circuitpython
Let me know what will be the correct way and I will use it.
Questions?
Would this imply to correct the simpletest_code also?. (I do not have all these hardware to test if after the changes, the library will work)

I would wait until guidance is given before continuing the update. Thanks

**Comment by dhalbert** [comment]
> I would wait until guidance is given before continuing the update.

We could discuss this In the Weeds in the weekly meeting tomorrow.

**Comment by jposada202020** [comment]
Yes good Idea 😃  I was hearing our meeting on feb 8th. regarding a similar topic.  That was regarding this issue https://github.com/adafruit/circuitpython/issues/4121. I will add this to the offtopic to the meeting thanks Dan

**Comment by jposada202020** [comment]
@dhalbert as discussed in our meeting I made the changes.  Hope everything is in there, if not please comment and I will correct accordingly. Thanks again for the support

**Comment by jposada202020** [comment]
After reflecting on this on the param descriptio we could use
```python
:param ~busio.I2C i2c_bus:`
```
when I use ~board.I2C it does not show as a Hyperlink in the documentation, do you know a way to make the correct reference?

**Comment by jposada202020** [comment]
@dhalbert I went through the Sphinx documentation `twice?` trying to find a solution for the parameter..no luck also tried different things, it seems that when we try to get the objects.inv from the documentation in readthedocs while building the api docs we could not get the correct information from the class objects. (I suppose, this is because board.I2C are not a class, could be way wrong). So I am returning to the parameter definition as ~busio.I2C. The examples and the readme.rst  for sure I would change to board.I2C().
Anyway if you have some time to review this, it would be really appreciate it, I would like to get this ball rolling over the weekend Thanks 😃

File: adafruit_icm20x.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ class ICM20X:  # pylint:disable=too-many-instance-attributes
 
 
     :param ~busio.I2C i2c_bus: The I2C bus the ICM20X is connected to.
-    :param address: The I2C slave address of the sensor
+    :param int address: The I2C slave address of the sensor
```
Comment (kattni) [inline adafruit_icm20x.py]: Please update the use of the word `slave` here to `peripheral` to match the CircuitPython Design Guide terminology.

**Review by kattni — CHANGES_REQUESTED** [review]
Please check through this and correct the use of `slave` to `peripheral` where used. Thanks!

File: adafruit_icm20x.py (hunk 1 above)
Comment (jposada202020) [reply adafruit_icm20x.py]: Will do thanks :)

Comment (jposada202020) [reply adafruit_icm20x.py]: I will go back and correct this and all the busio designations in previous PRs. Thanks @kattni

File: adafruit_icm20x.py [hunk 2]
```diff
@@ -139,7 +139,7 @@ class ICM20X:  # pylint:disable=too-many-instance-attributes
     """Library for the ST ICM-20X Wide-Range 6-DoF Accelerometer and Gyro Family
 
 
-    :param ~busio.I2C i2c_bus: The I2C bus the ICM20X is connected to.
+    :param ~board.I2C i2c_bus: The I2C bus the ICM20X is connected to.
```
Comment (jposada202020) [inline adafruit_icm20x.py]: @dhalbert The problem is here

Comment (jposada202020) [reply adafruit_icm20x.py]: Discussed in discord

File: adafruit_icm20x.py (hunk 1 above)
Comment (jposada202020) [reply adafruit_icm20x.py]: @kattni I updated this according to your suggestion and will continue to use the design guide for the comments

Comment (jposada202020) [reply adafruit_icm20x.py]: This is solved in https://github.com/adafruit/Adafruit_CircuitPython_ICM20X/pull/12/commits/5c27ce9680bb3ee3f51d3810496bea7ea2aab812

**Review by kattni — APPROVED** [review]

**Pull request merged**